Adds the export workouts to and import workouts from CSV feature

sgaamuwa opened the pull request.

sgaamuwa committed:
[#132976353] Adds Export Workout to CSV
Adds import and export buttons to overview.html
Adds export url to sub patterns for workouts in manager/urls.py
Adds an export_workout function to the workout.py file to print out to a CSV file

sgaamuwa committed:
Chore- Adds the settings file
Removes settings from the git ignore so that circle ci can pass

sgaamuwa committed:
[#132976353](Feature) Import Workouts from CSV
Adds import_workout function in workout.py to read CSV files and add them to the database.
Adds more fields to the exported CSV in export_workout in workout.py.
Adds url for CSV imports in wger/manager/urls.py.
Adds a form for adding a CSV to import from in wger/manager/overview.html
Adds migrations

sgaamuwa committed:
[#132976353](Chore) Settings file fix.
Sets dynamic setting variables in the settings.py file for circleci and moves it to the root directory

mfalade commented on wger/manager/views/workout.py:
```diff
@@ -17,27 +17,35 @@
 import logging
 import uuid
 import datetime
+import csv
```
Fix import order.

sgaamuwa committed:
[#132976353](Feature) Import and Export to CSV
Organises imports in workout.py
Improves code for exporting to CSV
Adds div for block options in overview.html and adds description for Importing

rolandgeider commented on wger/manager/views/workout.py:
```diff
@@ -215,6 +223,77 @@ def add(request):
+                one_set.exercises.add(
+                    Exercise.objects.filter(name=exercise).first()
+                )
+    return HttpResponseRedirect(reverse('manager:workout:overview'))
```
Wouldn't it make more sense to redirect to the newly imported workout? You can use the `get_absolute_url` method

sgaamuwa replied:
Yes, it would, but I had imagined that one might want to import more than one workout from the CSV file, so in that case I thought it best to redirect to the overview, so that they can see all the workouts imported, but I am open to ideas on how to improve this

rolandgeider replied:
Ah, of course, you are right. I had somehow understood it so that you could only export one workout, but you actually export all of them.

sgaamuwa committed:
[#132976353](Chore) Renaming urls and imports
Changes export to export_workout and import to import_workout in wger/manager/overview.html lines 37 and 40, and wger/manager/urls.py lines 57 and 60
Changes import order in wger/manager/views/workout.py line 25.
Separates the printed days and exercises using / and not - or * line 242, 248, 284 and 290

sgaamuwa committed:
Solves Merge conflicts

sgaamuwa committed:
Removes merge tags

sgaamuwa committed:
Removes Merge tags

sgaamuwa committed:
Adds Migrations for Merged branches

sgaamuwa committed:
Solves Merge conflicts and adds merge files
